lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -163,7 +166,6 @@ mlir::linalgx::mapToBRGEMMOp(RewriterBase &rewriter,
     FailureOr<SmallVector<Value>> maybeSlicedOperands =
         getSlicedOperands(builder, loc, localIvs, linalgOp, operandValuesToUse);
     if (failed(maybeSlicedOperands)) {
-      // TODO: is safe to just return{} ?
       assert(0 && "failed to generate loops");
```

Comment:
why not just `return maybeSlicedOperands` instead of this going around?

Reply:
Sorry what do you mean?

Reply:
If we fail to extract the slices when cannot build the loop nest and we assert.

Reply:
https://mlir.llvm.org/doxygen/structmlir_1_1linalg_1_1GenerateLoopNest.html

Reply:
maybe we can avoid failing. In fact, if we return failure something went really really wrong, what do you think?

Reply:
That's what I'm thinking... asserting and failing will both stop execution, but an assert is harder than a failure, and the latter may prove more user-friendly...

Reply:
Then either `return failure()` of some version with an error message would be more appropriate than `assert() && return {}`.

Reply:
I cannot propagate the failure, that my main problem. So basically the only option is to asset, as far as I can see reading the code in https://mlir.llvm.org/doxygen/structmlir_1_1linalg_1_1GenerateLoopNest.html

Reply:
ah! damn, ignore me, being thick is what's it. This code is in the lambda..